src/server/hotkey_collector.h

```diff
@@ -0,0 +1,73 @@
+//    was detected to be hot. The two types of hotkey, READ & WRITE, are detected
+//    separately.
+//
+//    +--------------------+  +----------------------------------------------------+
+//    |   Replcia server   |  | Hotkey collector                                   |
+//    |                    |  | +-----------------------------------------------+  |
+//    | +----------------+ |  | | Corase capture                                |  |
+//    | |                | |--> |                 +----------+                  |  |
+//    | |  RPC received  | || | |                 |   Data   |                  |  |
+//    | |                | || | |                 +-----+----+                  |  |
+//    | +-------+--------+ || | |                       |                       |  |
+//    |         |          || | |  +---------------+----v--+-------+---------+  |  |
+//    |         v          || | |  |       |Hot    |       |       |         |  |  |
+//    | +-------+--------+ || | |  |Bucket |Bucket |Bucket |Bucket |Bucket   |  |  |
+//    | |   Replication  | || | |  +-----------+-----------------------------+  |  |
+//    | | (only on the   | || | |              |                                |  |
+//    | |  write path))  | || | +--------------|--------------------------------+  |
+//    | +-------+--------+ || |             +--v---+                               |
+//    |         |          || |             | Data |                               |
+//    |         v          || |             +------+                               |
+//    | +-------+--------+ || |          +-----|-------+-------------+             |
+//    | |                | || |   +------|-------------|-------------|---------+   |
+//    | |  Capture data  ---| |   | Fine |capture      |             |         |   |
+//    | |                | |  |   |      |             |             |         |   |
+//    | +-------+--------+ |  |   | +----v----+   +----v----+   +----v----+    |   |
+//    |         |          |  |   | |  queue  |   |  queue  |   |  queue  |    |   |
+//    |         v          |  |   | +----+----+   +----+----+   +----+----+    |   |
+//    | +-------+--------+ |  |   |      |             |             |         |   |
+//    | |                | |  |   | +----v-------------v-------------v------+  |   |
+//    | |   Place data   | |  |   | |             Analsis pool              |  |   |
+//    | |   to the disk  | |  |   | +-----------------|---------------------+  |   |
+//    | |                | |  |   +-------------------|------------------------+   |
+//    | +----------------+ |  |                       v                            |
+//    |                    |  |                     Hotkey                         |
+//    +--------------------+  +----------------------------------------------------+
+
+class hotkey_collector
+{
+public:
+    // size: the cu size of raw_key/hash_key calculated by `capacity_unit_calculator`
```

Comment:
I  guess that size means the cu size of raw_key and value, but not the cu size of raw_key/hash_key